cmd/juju/commands/bootstrap.go

```diff
@@ -705,6 +705,9 @@ to create a new model to deploy %sworkloads.
 	}
 
 	isCAASController = jujucloud.CloudIsCAAS(cloud)
+	if isCAASController && c.ControllerCharmPath != "" {
+		return errors.NotSupportedf("deploying a local controller charm on a k8s controller")
```

Comment:
How about IAAS?

Reply:
works fine for IAAS since we upload the local charm to the machine and use it (like we do for custom built agent)